src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,177 @@
+'use client';
+import React, { useState, useRef, useEffect } from 'react';
+import SliderNav from '../SliderNav/SliderNav';
+import SectionHeader from '../SectionHeader/SectionHeader';
+import { ArrowUpRight } from 'lucide-react';
+import './Slider.scss';
+interface SliderProps {
+  sliderName: string;
+  items: { imageUrl: string; description: string; year: number }[];
+}
+
+export default function Slider({ sliderName, items }: SliderProps) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [startX, setStartX] = useState(0);
+  const [isDragging, setIsDragging] = useState(false);
+  const [dragOffset, setDragOffset] = useState(0);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const [isFading, setISFading] = useState<boolean>(true);
+  const handleDragStart = (
+    e: React.MouseEvent<HTMLDivElement> | React.TouchEvent<HTMLDivElement>,
+  ) => {
+    setIsDragging(true);
+    if ('touches' in e) {
+      setStartX(e.touches[0].pageX);
```

Comment:
Kindly use null check operator here e?.touches[0]?.pages, similarly for e?.pageX